src/Timecode.php

```diff
@@ -0,0 +1,475 @@
+<?php
+
+namespace FireWorkWeb\SMPTE;
+
+/**
+ * Timecode class representation
+ */
+class Timecode
+{
+    /**
+     * @var float
+     */
+    const DEFAULT_FRAMERATE = 24.0;
+
+    /**
+     * @var bool
+     */
+    const DEFAULT_DROPFRAME = false;
+
+    /**
+     * @var array
+     */
+    const SUPPORTED_FRAMERATES = [
+        23.976,
+        24,
+        25,
+        29.97,
+        30,
+    ];
+
+    /**
+     * @var int
+     */
+    const DROP_FRAME_CONCEPT = 17982;
+
+    /**
+     * @var float
+     */
+    private $frameRate;
+
+    /**
+     * @var bool
+     */
+    private $dropFrame;
+
+    /**
+     * @var int
+     */
+    private $frameCount;
+
+    /**
+     * @var int
+     */
+    private $hours;
+
+    /**
+     * @var int
+     */
+    private $minutes;
+
+    /**
+     * @var int
+     */
+    private $seconds;
+
+    /**
+     * @var int
+     */
+    private $frames;
+
+    /**
+     * Create a new Timecode
+     *
+     * @param int|string|DateTime $time
+     * @param float $frameRate
+     * @param bool $dropFrame
+     * @return void
+     */
+    public function __construct($time, float $frameRate = self::DEFAULT_FRAMERATE, bool $dropFrame = self::DEFAULT_DROPFRAME)
+    {
+        if (! self::isValidTimeCode($time, $frameRate, $dropFrame)) {
+            throw new \InvalidArgumentException('Invalid timecode');
+        }
+
+        $this->frameRate = $frameRate;
+        $this->dropFrame = $dropFrame;
+
+        $this->generateFrameCount($time);
+    }
+
+    /**
+     * Determine if is valid timecode
+     *
+     * @param int|string|DateTime $time
+     * @param float $frameRate
+     * @param bool $dropFrame
+     * @return self
+     */
+    public static function isValidTimeCode($time, float $frameRate, bool $dropFrame)
+    {
+        if (! self::isFrameRateSupported($frameRate, $dropFrame)) {
+            throw new \InvalidArgumentException('Frame rate not supported');
+        }
+
+        return self::isValidInteger($time) || self::isValidDate($time) || self::isValidString($time, $frameRate, $dropFrame);
```

Comment:
let's break this line

```suggestion
        return self::isValidInteger($time)
            || self::isValidDate($time)
            || self::isValidString($time, $frameRate, $dropFrame);
```

Reply:
What do you guys think about to create a `Concern` for these validations? Or a single class?

```php
// e.g
use Concerns\Validations

// e.g
Validations::isValidInteger();
```

Reply:
@wesleyhf 

We need to export those validations.. If we can do a way to use them and also export, then great 👍